src/downloadpanel.cpp

```diff
@@ -96,17 +87,23 @@ QNetworkReply *DownloadPanel::getRepliesListItem( int index )
     return replies.at( index );
 }
 
-#define FIND_SENDER_INDEX \
-int senderIndex = -1; \
-for( int i = 0; i < replies.size(); i++ ) \
-    if( replies[ i ] == QObject::sender() ) { \
-        senderIndex = i; \
-        break; }
+int DownloadPanel::findSenderIndex( QObject *theSender )
+{
+    int senderIndex = -1;
+
+    for( int i = 0; i < replies.size(); i++ )
+        if( replies[ i ] == theSender ) {
+            senderIndex = i;
+            break;
+    }
+
+    return senderIndex;
+}
 
 void DownloadPanel::networkReplyProgress(
     qint64 bytesReceived, qint64 bytesTotal )
 {
-    FIND_SENDER_INDEX
+    int senderIndex = findSenderIndex( QObject::sender() );
```

Comment:
Try this line instead of findSenderIndex?
`int senderIndex = replies.indexOf(reinterpret_cast<QNetworkReply *>(QObject::sender()));`
The `replies[ i ] == theSender` does the same kind of cast in findSenderIndex anyway.